sources/HUBViewControllerImplementation.m

```diff
@@ -392,7 +392,8 @@ - (CGRect)frameForBodyComponentAtIndex:(NSUInteger)index
     }
     
     NSIndexPath * const indexPath = [NSIndexPath indexPathForItem:(NSInteger)index inSection:0];
-    return [self.collectionView layoutAttributesForItemAtIndexPath:indexPath].frame;
+    HUBCollectionViewLayout * const layout = (HUBCollectionViewLayout *)self.collectionView.collectionViewLayout;
```

Comment:
I could skip the casting, I just added it for clarity to understand that it differs from a UICollectionViewLayout. 

If you call `layoutAttributesForItemAtIndexPath` on a collectionView directly, the collection view asserts if there is no layout for that indexpath.

But in our own `HUBCollectionViewLayout` class, we do this: 
```objc
- (nullable UICollectionViewLayoutAttributes *)layoutAttributesForItemAtIndexPath:(NSIndexPath *)indexPath
{
    return self.layoutAttributesByIndexPath[indexPath];
}
```

which just returns nil. 
And this method can be called before the layout is done if called while the app is in the background. By using our method, we just return nil instead of asserting.